services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/llm_prompt_builder.rb

```diff
@@ -4,21 +4,53 @@ module Evidence
   module Research
     module GenAI
       class LLMPromptBuilder < ApplicationService
+        class InvalidLLMPromptTemplateIdError < StandardError; end
+        class InvalidPassagePromptIdError < StandardError; end
+
+        DELIMITER = "~~~"
```

Comment:
You might consider some comments in this file. I probably should have looked at the tests sooner (and I might be a little tired), but it took me a bit to work through where the `Regexp.last_match(1)&.to_i` line in the `prompt` code. A comment would have been clarifying here, e.g.
```ruby
# Sample usage of delimiter in an LLMPromptTemplate
# Insert prompt:
# ~~~prompt~~~
# Insert 20 examples:
# ~~~examples,20~~~

Reply:
Good idea.  Regex can be hard to grok.